app/controllers/records_controller.rb

```diff
@@ -2,6 +2,7 @@ class RecordsController < ApplicationController
   include RecordsControllerBehavior
 
   def ingest_options
+    authorize! :new, GenericAsset
```

Comment:
I did that because the built-in auth from RecordsControllerBehavior only protects known actions.  Anything we add, we have to protect manually.  I could have added a filter, but to avoid double-checking with cancan, it would have been a filter that only runs on `:ingest_options`, which seems like that could be weird.